### PR TITLE
Integrate add appointment

### LINKED ISSUE #16: Integrate add appointment with backend
# Context

* The button add appointment is ready, but not functional. We need to make it functional, so users can add appointments in the sistem.

# Tech planning

*tbd

# Acceptance criteria

* Users can add appointments in the sistem.

### PR DESCRIPTION
**Pull request by IagoGoncalves22**
Resolves #16 

## Problem

We don't have an integrated add appointment modal yet.

## Changes proposed in this PR

1. Integreted add appointment modal with backend Firebase.
2. Improved file src/api/addAppointments.ts.
3. Improved file src/components/AddAppointmentModal.tsx.
4. Utlized format, Yup and formik to improve types and UX.
5. In this issue also was improved edit appointment modal and edit customer modal utilizing formik.
6. In this issue also was improved delete appointment and edit appointment utilizing replace component.
7. Reviwed and improve translation of all appointment and customer modals.

### CODE REVIEW

File: src/api/fetchCustomers.ts [hunk 1]
```diff
@@ -17,3 +18,29 @@ export const fetchCustomers = async (companyId: string) => {
...
+    const customerRef = collection(db, `companies/${companyId}/customers`);
+    const querySnapshot = await getDocs(
+        query(
+            customerRef,
+            where('customerName', '>=', search),
+            where('customerName', '<=', search + '~')
+        )
+    );
+
+    // const q2 = query(customerRef, where("state", "==", "CA"), where("population", "<", 1000000));
```
Comment (galexandrade) [inline src/api/fetchCustomers.ts:35]: I think we can remove this.

File: src/components/EditAppointmentModal.tsx [hunk 2]
```diff
@@ -31,8 +33,38 @@ const EditAppointmentModal = ({
...
-    const [time, setTime] = useState(appointment.time);
+    const schema = Yup.object().shape({
+        day: Yup.date().required(t('scheduleRow.schemaDay')),
+        time: Yup.string().required(t('scheduleRow.schemaTime'))
+    });
+
+    const formik = useFormik({
+        initialValues: {
+            day: new Date(),
+            time: ''
```
Comment (galexandrade) [inline src/components/EditAppointmentModal.tsx:44]: These fields need to be the current values so they can bring the current data to the screen:
image

File: src/components/EditCustomerModal.tsx [hunk 3]
```diff
@@ -23,51 +25,65 @@ type Props = {
...
     const [isLoading, setIsLoading] = useState(false);
     const appContext = useAppContext();
+    const schema = Yup.object().shape({
+        customerName: Yup.string().required(t('customerRow.schemaCustomer')),
+        nickName: Yup.string().required(t('customerRow.schemaNickname'))
+    });
+
+    const formik = useFormik({
+        initialValues: {
+            customerName: '',
```
Comment (galexandrade) [inline src/components/EditCustomerModal.tsx:35]: Same for these ones. The initial values should come from the data that is already saved for the customer:
image

**Review by galexandrade — COMMENTED** [review]
It looks great. Just noticed the edit modals are not loading the initial data in the form.

File: src/api/fetchCustomers.ts (hunk 1 above)
Comment (IagoGoncalves22) [reply src/api/fetchCustomers.ts:35]: Right. It was deleted.

File: src/components/EditCustomerModal.tsx (hunk 3 above)
Comment (IagoGoncalves22) [reply src/components/EditCustomerModal.tsx:35]: Great. It was corrected.

File: src/components/EditAppointmentModal.tsx (hunk 2 above)
Comment (galexandrade) [reply src/components/EditAppointmentModal.tsx:44]: You would need to populate the `initialValues`:

```
initialValues: {
    day: appointment.day,
    time: appointment.time
}
```

Comment (IagoGoncalves22) [reply src/components/EditAppointmentModal.tsx:44]: Thanks for your help. It worked.

**Review by galexandrade — APPROVED** [review]
It looks great 🎉

**Pull request merged**